esmvaltool/preprocessor/_download.py

```diff
@@ -60,21 +61,40 @@ def synda_download(synda_name, dest_folder):
             'synda', 'get', '--dest_folder={}'.format(dest_folder),
             '--verify_checksum', synda_name
         ]
-        cmd = ' '.join(cmd)
-        logger.debug("Running: %s", cmd)
-        subprocess.check_call(cmd, shell=True)
+        logger.debug("Running: %s", ' '.join(cmd))
+        subprocess.check_call(cmd)
 
     return local_file
 
 
-def download(files, dest_folder):
+def download(files, dest_folder, variable, rootpath, drs,
+             to_repository, compress_downloads):
     """Download files that are not available locally"""
+    if to_repository:
```

Comment:
computing the dest_folder should be done in _recipe.py, changing it inside this function is very confusing.

Reply:
I agree. I should have spent a few minutes cleaning the code. Sorry!